src/core_plugins/kibana/public/visualize/listing/visualize_landing_page_table.js

```diff
@@ -71,17 +45,11 @@ export class VisualizeLandingPageTable extends React.Component {
   }
 
   onFilter = (newFilter) => {
-    const PAGE_SIZE = 3;
+    this.setState(filterItems());
```

Comment:
This isn't very readable imo, because I'd see this and think, why am I filtering here?  What do you think of something like this:

```
this.setState(ListingTableActions.startFetching());
 this.props.fetch(newFilter).then((results) => {
      const items = results.hits;
		  const pagesCount = this.pager.getPagesCount(items.length);
      this.setState(ListingTableActions.setFilteredItems(this.state, items, pagesCount));
}.finally(() => {
  this.setState(ListingTableActions.endFetching());
}
```
This also gives us a place to show any errors.  

What did you think of @weltenwort's suggestion of passing in setState?  I ask because it could simplify this refactor if we passed in that function.  Doesn't have to be tied to React, all the Actions class needs to know is to call the function with the new state.

Then we could further refactor to:
```
ListingTableActions.filter(this.setState, this.state, this.props.fetch, this.pager); 
```
or do you think that is too abstracted?